src/signet.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <signet.h>
+
+#include <consensus/merkle.h>
+#include <consensus/params.h>
+#include <hash.h>
+#include <primitives/block.h>
+#include <primitives/transaction.h>
+#include <script/interpreter.h>
+#include <script/standard.h>        // MANDATORY_SCRIPT_VERIFY_FLAGS
+#include <util/system.h>
+
+int GetWitnessCommitmentIndex(const CBlock& block);
+template<typename T> int GetWitnessCommitmentIndex(const T& tx);
+
+static constexpr int NO_WITNESS_COMMITMENT{-1}; // note: this is copied from validation.h, to avoid a circular dependency issue
```

Comment:
In commit 7168395ea63d33075a60369f5778eb635208beba

I don't think we solve circular dependencies by copy pasting parts of the dependencies into each other. Doing so slightly defeats the purpose of having separate modules. There are at least two options:

* Ignore the circular dependency for now (and document it as an exception in the linter)
* Move the witness commitment stuff to a new "module". I think libconsensus `consensus/witness_commitment` (or similar) could make sense. This would also help the miner not depend on the server (in a theoretical world where the miner is a separate util binary, where all state is passed in), but only on libconsensus.

Reply:
Moving to avoid duplicate makes sense; honestly not sure what I was thinking here.

I'm not sure it warrants a new module, so I've put it in `consensus/validation.h` for now. If you really do think it should be in a separate file, I'll move it.